src/mds/Server.cc

```diff
@@ -3839,7 +3839,8 @@ void Server::handle_client_setattr(MDRequestRef& mdr)
 								   truncating_smaller, changed_ranges));
 
   // flush immediately if there are readers/writers waiting
-  if (cur->get_caps_wanted() & (CEPH_CAP_FILE_RD|CEPH_CAP_FILE_WR))
+  if (xlocks.count(&cur->filelock) &&
+      (cur->get_caps_wanted() & (CEPH_CAP_FILE_RD|CEPH_CAP_FILE_WR)))
```

Comment:
Why do we only want to flush if *we* are xlocking? get_caps_wanted() is telling us that *somebody* wants Frw caps they don't have yet, and we want to service them, right?